### PR TITLE
Paramatch has been Mungebittified

### PR DESCRIPTION
**Pull request by bruno615**
Wrote new tests, updated documentation.  Good to go!

### CODE REVIEW

**Comment by bruno615** [comment]
@robertzk Added blacklisting capability.  This is ready for review.

File: R/paramatch.r [hunk 1]
```diff
@@ -1,49 +1,73 @@
...
     frequency <- data.frame(word = words, 
                             occurances = sapply(words, function(x) length(grep(paste0("\\<",x,"\\>"), allwords))), 
                             stringsAsFactors = FALSE) 
-  # Ensure that N isn't greater than the total number of unique words
+    # Blacklist
+    if (length(blacklist) >0) {
+      for (i in 1:length(blacklist)) {
+        frequency <- frequency[!frequency$word == blacklist[[i]],]
+      }
+    }
```
Comment (rylandely) [inline R/paramatch.r]: you can also replace the whole block with something like 
`frequency <- frequency[!frequency$word %in% blacklist]`

File: R/paramatch.r [hunk 2]
```diff
@@ -1,49 +1,73 @@
...
       top_n_words <- length(words)
     }
-  # Take the top N words
-    top_n <- frequency[order(-frequency$occurances),][1:top_n_words,]
-  # Create the output data frame
-    output <- data.frame(input, sapply(top_n$word,
-      function(x) stingr::str_count(paragraph, paste0("\\<",x,"\\>"))))
-    colnames(output) <- c("input", sapply(top_n$word, function(x) paste0("col_", x)))
+    # Take the top N words
+    top_n <- frequency[order(-frequency$occurances),][1:top_n_words, 1]
```
Comment (rylandely) [inline R/paramatch.r]: alternatively, you can use the `decreasing = TRUE ` parameter to `order(...)`

File: R/paramatch.r (hunk 1 above)
Comment (bruno615) [inline R/paramatch.r]: That's what I was looking for!

File: R/paramatch.r (hunk 2 above)
Comment (bruno615) [inline R/paramatch.r]: That improves readability, will incorporate

File: R/paramatch.r [hunk 3]
```diff
@@ -1,49 +1,73 @@
...
-    stopifnot(is.logical(suppress.input))
+  stopifnot(is.character(paragraph_col))
+  stopifnot(is.numeric(top_n_words))
+  stopifnot(top_n_words >= 1)
+  stopifnot(top_n_words %% 1 == 0)
+  stopifnot(is.logical(suppress.input))
   # Standardize the input
-    paragraph <- toupper(input)
-    paragraph <- gsub("[[:punct:]]", " ", paragraph)
-    paragraph <- gsub("[[:space:]]+", " ", paragraph)
```
Comment (rylandely) [inline R/paramatch.r]: If you wanna get really fancy you can try to find a package to reduce various cases/tenses to a single common word stem, e.g. the tm package: http://cran.r-project.org/web/packages/tm/tm.pdf

Comment (bruno615) [inline R/paramatch.r]: That's a good idea for the next update

File: R/paramatch.r [hunk 4]
```diff
@@ -1,49 +1,69 @@
...
+    # Write top_n to the Environment
+    inputs$top_n <<- top_n
+  } else {
+    top_n <- inputs$top_n
+  }
 
-  output[if(suppress.input) {-1} else {TRUE}]
+  # Run the count for the top n words
+  output <- data.frame(sapply(top_n, function(x) stringr::str_count(paragraph_col, paste0("\\<", x, "\\>"))))
+  colnames(output) <- c(sapply(top_n, function(x) paste0("col_", x)))
```
Comment (robertzk) [inline R/paramatch.r]: Don't need the `c` wrapper.` Also, can you replace all instances of `sapply` with `vapply` (last argument `character(1)`)

File: R/paramatch.r [hunk 5]
```diff
@@ -1,49 +1,69 @@
...
 #' @param suppress.input logical.  Suppresses the Input column of the output data frame.
+#' @param blacklist character. Prevents undesired words participating int he output.
 #' @return numeric. Number of occurances of the particular word in each line.
-#' @export
+#' @author Mike Bruno
 #' @examples
-#' paragraph <- c("This is an example.", "Great code uses examples like this example.")
-#' paramatch(paragraph, 2)
-paramatch <- function(input, top_n_words = 5, suppress.input = FALSE) {
+#' paragraph <- data.frame(id = c(1:2), text = c("This is an example.", "Great code uses examples like this example."), stringsAsFactors = FALSE)
```
Comment (robertzk) [inline R/paramatch.r]: Need to fix this example to use a `mungebit`, or it breaks.

**Comment by robertzk** [comment]
Good to go after you fix the example and CI passes

**Comment by bruno615** [comment]
@robertzk This is good to go

**Pull request merged**